src/tracker/face_id/identifier.py

```diff
@@ -130,31 +135,38 @@ def compute_known_embeddings(self):
             label_path = os.path.join(path_to_known_faces, directory)
             embeddings = []
             for file in os.listdir(label_path):
-                if (
-                    (".jpg" in file.lower())
-                    or (".jpeg" in file.lower())
-                    or (".png" in file.lower())
-                ):
-                    im = Image.open(label_path + "/" + file).convert(
-                        "RGB"
-                    )  # convert in RGB because png are RGBA
-                    img_array = np.array(im)
-                    uint8_tensor = (
-                        torch.from_numpy(img_array).permute(2, 0, 1).contiguous()
-                    )
-                    float32_tensor = uint8_tensor.to(dtype=torch.float32)
-
-                    float32_tensor = float32_tensor.to(self.device)
-                    face = self.detector.extract_face(float32_tensor)
-                    if self.debug:
-                        save_tensor(face, f"{directory}.jpeg")
-                    # TODO: check if there is only one face here
-                    embed = self.feature_extractor.get_embedding(face)
-                    embeddings.append(embed)
-                else:
+                if not any(file.lower().endswith(suffix)
+                           for suffix in (".jpg", ".jpeg", ".png")):
                     LOGGER.warning(
                         "Ignoring unsupported file type: %s. Only .jpg, .jpeg, and .png files are supported.",  # pylint: disable=line-too-long
                         file,
                     )
-
+                    continue
+
+                im = Image.open(label_path + "/" + file).convert(
+                    "RGB"
+                )  # convert in RGB because png are RGBA
+                img_array = np.array(im)
+                uint8_tensor = (
+                    torch.from_numpy(img_array).permute(2, 0, 1).contiguous()
+                )
+                float32_tensor = uint8_tensor.to(dtype=torch.float32)
+
+                float32_tensor = float32_tensor.to(self.device)
+                face = self.detector.extract_face(float32_tensor)
+                if face is None:
+                    LOGGER.warning(f"Unable to find face in {directory}/{file}")
+                    continue
+
+                if self.debug:
+                    save_tensor(face, f"{directory}.jpeg")
+                # TODO: check if there is only one face here
+                embed = self.feature_extractor.get_embedding(face)
+                embeddings.append(embed)
+
+            if not embeddings:
+                raise NoFacesDetectedError(
+                    f"Unable to recognize any face for supposedly-known "
```

Comment:
Same here -avoid using the word "embedding". I would say "Unable to recognize any face in pictures from {directory}. Are you sure it contains a picture of a person's face?"